walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,92 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;  
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    ArrayList<TimeRange> possibleTimes;
+
+    public FindMeetingQuery() {
+        possibleTimes = new ArrayList<TimeRange>();
+    }
+
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        long reqDuration = request.getDuration();
```

Comment:
Reset possibleTimes here to an empty list, in case this method is called multiple times with the same FindMeetingQuery object.